### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by brettp**
Added warning that this project is unmaintained and does not work with current hardware or software.

Please add this to the readme or make the repo private so people don't see it and try to use it.

### CODE REVIEW

**Comment by marcosscriven** [comment]
Hi @brettp 

Thanks for the suggestion here. I'm happy to update to say it's unmaintained, and that it's never been tested on the latest Google Nest products.

However, I see no reason it shouldn't work with some updates and therefore doesn't seem to make sense to take down.

Unfortunately I no longer have one of these devices to test it with.

EDIT - Just checked the code, and the developer switch check is still there https://chromium.googlesource.com/chromiumos/third_party/coreboot/+/refs/heads/firmware-gale-8281.B/src/mainboard/google/gale/chromeos.c#78

**Comment by AptGetGnome** [comment]
Hi @marcosscriven 

I'm still using the older model of Google Wifi, and was considering rooting to be able to change the DNS settings for the purpose of Pi-Hole without making it my DHCP server, would you believe, in theory that my system can be rooted?
Model: NLS-1304-25
Software: 12871.57.12

**Comment by brettp** [comment]
@marcosscriven It's not switching into developer mode that's the problem; it's being able to flash a working image. Google have started checking signing keys for updates and drivers, so custom firmware is not able to start networking. This isn't something that is likely to change, so these custom images will not be applicable unless an exploit is found.

@adamsarin No, this will not work with old hardware unless you haven't connected it to the internet....which would pretty much defeat the purpose of the hardware. It auto updates the firmware to a version that breaks this as soon as you register it with Google. There is no reliable way to revert to a state that allows custom firmware to work.

**Comment by AptGetGnome** [comment]
@brettp Thanks for your reply, is there any current roots for modern firmware or has it all been abandoned?

**Comment by brettp** [comment]
None that I'm aware.

**Comment by marcosscriven** [comment]
Again, I see no reason why it shouldn't work: see https://www.chromium.org/chromium-os/developer-information-for-chrome-os-devices/custom-firmware

> In developer mode, the RW firmware doesn't verify that the kernel is signed by Google, just that it looks like a correctly signed kernel image. It doesn't matter who signed it, so you can build and boot your own kernel and rootfs.

There seems to be plenty of interest in this - just needs someone who still actually has one.

**Comment by jmsgomes** [comment]
I have multiple TP-Link Google OnHub devices - I was hoping to eventually free one up from my current mesh to try this out. Please let me know if this would work on the TP-Link Google OnHub and, once I get some free time this year, I could try to build, sign and install this to see how it works on that model.

**Comment by brettp** [comment]
> Again, I see no reason why it shouldn't work: see https://www.chromium.org/chromium-os/developer-information-for-chrome-os-devices/custom-firmware
> 
> > In developer mode, the RW firmware doesn't verify that the kernel is signed by Google, just that it looks like a correctly signed kernel image. It doesn't matter who signed it, so you can build and boot your own kernel and rootfs.
> 
> There seems to be plenty of interest in this - just needs someone who still actually has one.

The problem is not flashing or booting -- Network drivers do not load on custom firmware anymore. This is deliberate on Google's side. If you want to keep the repo up for posterity that is of course fine, but it needs to clearly not be presented as a working solution.

**Comment by marcosscriven** [comment]
@brettp - thanks for the explanation. That is indeed sad news. I've updated the readme accordingly.

**Comment by vapier** [comment]
> @marcosscriven It's not switching into developer mode that's the problem; it's being able to flash a working image. Google have started checking signing keys for updates and drivers, so custom firmware is not able to start networking. This isn't something that is likely to change, so these custom images will not be applicable unless an exploit is found.

what are you referring to ?  if you're booting custom firmware, you have a custom kernel, and you can load whatever drivers you want.  only the kernel is responsible for checking its own modules.

as for updates, that's how CrOS has always worked -- any update it processes has to be signed.  however, the key used to sign the updates is embedded in the rootfs, so you can control it with your custom firmware.  if you expect to update an official firmware to your own custom one, that would never work.

**Comment by forresthopkinsa** [comment]
Subscribing to this issue to see if there are any further developments on this...

**Comment by marcosscriven** [comment]
@brettp - Further to @vapier's question, I've reverted the change for now. I'd be interested in your clarification too.

I was under the impression you had tried this and established it could no longer work.

**Comment by brettp** [comment]
I did try this, which is why I made the original PR. I no longer have the dev env set up to give the full details. If I get free time soon I will try again.

The tl;dr is this repo does not work as is, and the README should reflect that. If someone can make PRs to fix it, that would be great, but theorizing that it _should_ work is a long way away from it actually working, and is not a basis for describing that it _does_ work. 

Here are the details I remember:

1. Building the images works correctly
2. Switching the pucks to dev mode works correctly
3. Flashing the images works correctly
4. The router's network interfaces never come up, neither wifi nor lan. There were errors about loading the binary blobs, but I can't remember specifics.
5. There was something wrong with the busybox binary that required manual intervention. Don't remember exactly what.

In addition, other [tickets](https://github.com/marcosscriven/galeforce/issues/40) indicate that even if it can load drivers, as soon as it connects to the internet Google will overwrite the firmware.

If no one has the time and hardware to fix this, why contribute to incorrect information with outdated documentation in the README?

**Comment by marcosscriven** [comment]
@brettp - I'm certainly happy to state it's unmaintained.

My concern specifically is the notion it is fundamentally no longer possible, which I think would also be misleading.

**Comment by peterleiser** [comment]
@marcosscriven @brettp @adamsarin @jmsgomes @vapier 

**### TLDR; New Galeforce installs are DEFINITELY working!**  

I'm currently connected to the Google WiFi device that I just flashed today.  It has a working external internet connection and that is the internet connection that I am using right now while connecting to it over wifi.  I used the gale.bin.tar.gz prebuilt binary from tag 0.5: https://github.com/marcosscriven/galeforce/releases/tag/0.5

My device Model is AC-1304, and I can reliably ssh as root to the device.

I have never used Galeforce before today because the comments here indicated that it was no longer possible.  However, it is absolutely possible because I am connected over 5 GHz Wifi to that Galeforce-flashed Google WiFi device right now.

**[Steps to reproduce]**

I have two routers: (1) main router is a Linksys WRT3200ACM with OpenWrt (2) Google Nest WiFi router that is on the same 192.168.1.0 subnet as the main router is has the LAN port connected to the main router LAN (yes, it is possible, you can read below if you want).

I had my laptop connected to the LAN port of the flashed device, and the WAN port of the flashed device was connected to the LAN port of my main router. After flashing the device I was able to ssh into it over the LAN port, but there was no internet connection (couldn't even ping the main router).  I was also able to connect to the default setupXXXXX ssid.  So I disconnected the ethernet cable from the laptop and connected the LAN port of the flashed device to my main router.  There was still no internet connection.

Then I tried to add the flashed Google WiFi point to my existing Google Nest WiFi router.  It was not added to the Google Nest WiFi mesh (it failed the setup at the "adding to network" or "finalizing mesh" step), but then internet access started working from the flashed device.  However, I am only able to connect WiFi to that default setupXXXXX ssid, and that setupXXXXX ssid broadcasts on both 2.4 GHz and 5 GHz.

But fundamentally it works.  The networking drivers for ethernet and wifi are obviously working, it has an internet connection, and you can connect to it over wifi and have an internet connection.  One caveat is that the LAN of the flashed router is still 192.168.84.0, so in order to ssh into it I have to connect to the setupXXXXX ssid and set the DHCP to manual IP (I use 192.168.84.100).

If you have any questions about it, please don't hesitate to ask me.

Cheers!
Peter

**[Details about the OpenWRT router and the Google Nest Router being on the same 192.168.1.0 subnet]**

I have a Google Nest WiFi router (with a different ssid than my OpenWrt main router) that has the LAN port connected to my main router LAN,  and the WAN port is connected to a VLAN port on the main router that has a separate subnet (Nest Wifi doesn't work if you don't allow the router to have a WAN connection).  The Nest Wifi provides DHCP only to the other 4 Google WiFi and Nest WiFi points in the mesh, and has a DHCP pool of size 4 so that only those points get an IP address from that DHCP.  I configured the dnsmasq on the OpenWrt router to ignore the MAC addresses of the Google Nest and WiFi devices.  The Google Nest WiFi router is configured to have an IP address (192.168.1.3) on the same subnet as the LAN, and any devices that connect to either the OpenWrt ssid or the Google Nest ssid are on the same 192.168.1.0 subnet (they fall through to using the DHCP from the main router)  and all devices can see and talk to each other.  Just trust me, it works, and I had my reasons. ;-) If you want to know how I did it let me know....

**Comment by jmsgomes** [comment]
>The Nest Wifi provides DHCP only to the other 4 Google WiFi and Nest WiFi points in the mesh, and has a DHCP pool of size 4 so that only those points get an IP address from that DHCP. I configured the dnsmasq on the OpenWrt router to ignore the MAC addresses of the Google Nest and WiFi devices. The Google Nest WiFi router is configured to have an IP address (192.168.1.3) on the same subnet as the LAN, and any devices that connect to either the OpenWrt ssid or the Google Nest ssid are on the same 192.168.1.0 subnet (they fall through to using the DHCP from the main router) and all devices can see and talk to each other. Just trust me, it works, and I had my reasons. ;-) If you want to know how I did it let me know....

Can you confirm all you did to get the regular clients (non-Google/Nest WiFi devices) to always get leases from a different DHCP server (ignoring DHCP from Google/Nest Wifi)? I haven't minded too much letting Google/Nest Wifi run its own show with one exception: serving DHCP (which mediately controls DNS and routing), which I'd really prefer to control with my own router (I have an Ubiquiti ER-12)

**Comment by peterleiser** [comment]
> Can you confirm all you did to get the regular clients (non-Google/Nest WiFi devices) to always get leases from a different DHCP server (ignoring DHCP from Google/Nest Wifi)? I haven't minded too much letting Google/Nest Wifi run its own show with one exception: serving DHCP (which mediately controls DNS and routing), which I'd really prefer to control with my own router (I have an Ubiquiti ER-12)

I followed this method from Steven Campbell: https://support.google.com/googlenest/thread/832837?hl=en

**Comment by peterleiser** [comment]
@jmsgomes If you have any questions please don't hesitate to ask.  And I won't be critical or condescending either. :-)  I see a lot of bad attitudes on the OpenWrt forums and it's very disappointing.  

The biggest problem that I experienced while trying to get Nest Wifi working on the same subnet as the main router was that the WAN port of the Nest Wifi router must be plugged in the entire time. I definitely found that having the WAN port connected to my main router's regular 192.168.1.1 LAN was no bueno.   What worked for me was is to have the WAN connected to a VLAN port on the main router that has it's own subnet and DHCP, and that cannot talk to the main router LAN subnet.  Before that I also had it working by connecting the WAN port to an ATT MiFi router.  After everything is setup and the Nest Wifi LAN is on the same subnet as the main router the devices will use the main router LAN, and the Nest Wifi WAN port will only be used for Nest Wifi updates, connections from the mobile app, and speed tests.

**Comment by jmsgomes** [comment]
> @jmsgomes If you have any questions please don't hesitate to ask.  And I won't be critical or condescending either. :-)  I see a lot of bad attitudes on the OpenWrt forums and it's very disappointing.  

Much appreciated :-D

> The biggest problem that I experienced while trying to get Nest Wifi working on the same subnet as the main router was that the WAN port of the Nest Wifi router must be plugged in the entire time. I definitely found that having the WAN port connected to my main router's regular 192.168.1.1 LAN was no bueno.   What worked for me was is to have the WAN connected to a VLAN port on the main router that has it's own subnet and DHCP, and that cannot talk to the main router LAN subnet.  Before that I also had it working by connecting the WAN port to an ATT MiFi router.  After everything is setup and the Nest Wifi LAN is on the same subnet as the main router the devices will use the main router LAN, and the Nest Wifi WAN port will only be used for Nest Wifi updates, connections from the mobile app, and speed tests.  

Thank you for these pointers. I'm looking forward to trying this out this weekend or the next...

**Comment by peterleiser** [comment]
@jmsgomes Hi Jorge. Did you successfully get Google/Nest WiFi working with a different DHCP server?

**Comment by abo-github** [comment]
I wonder if anyone has ever tried to write or create their own firmware for Google WiFi based on OpenWRT ... or whether this is even possible.

**Comment by ricardojoaoreis** [comment]
Leaving here my experience for whoever is unsure whether this works or not

Like @peterleiser mentioned, flashing `gale.bin` tagged with 0.5 works, to an extent... I'm able to SSH into the puck but as soon as I try to setup a new network it updates and starts blinking red, which means factory reset and SSH no longer works.

Building a new image using the script works (after updating dropbear and busybox URLs), the device stays in developer mode and the network interfaces are up unlike what @brettp says. Running an nmap scan on 192.168.84.1 I get this:
```
$ nmap 192.168.84.1
Starting Nmap 7.91 ( https://nmap.org ) at 2021-02-04 20:28 WET
Nmap scan report for testwifi.here (192.168.84.1)
Host is up (0.0062s latency).
Not shown: 997 closed ports
PORT     STATE SERVICE
53/tcp   open  domain
80/tcp   open  http
5000/tcp open  upnp
```
So SSH is blocked, however this is where I can't progress further. From my limited understanding this might be because there's a firewall on the puck blocking SSH or maybe dropbear is failing to start... Any tips @marcosscriven?

**Comment by brettp** [comment]
@ricardojoaoreis - No need to tag me as I've moved on to more suitable, reliable, and open hardware. Best of luck.
